src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -10310,6 +10310,53 @@ and TcAndBuildFixedExpr (cenv: cenv) env (overallPatTy, fixedExpr, overallExprTy
 
     let g = cenv.g
 
+    // Search for GetPinnableReference (like https://learn.microsoft.com/en-us/dotnet/api/system.span-1.getpinnablereference?view=net-7.0)
+    // on the target expression, and, if it exists, call it
+    let tryBuildGetPinnableReferenceCall () =
+        let getPinnableReferenceMInfo =
+            TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AllResults cenv env mBinding env.eAccessRights "GetPinnableReference" overallExprTy
+            |> List.tryFind (fun mInfo ->
+                // GetPinnableReference must be a parameterless method with a byref or inref return value
+                match mInfo.GetParamDatas(cenv.amap, mBinding, mInfo.FormalMethodInst), mInfo.GetFSharpReturnType(cenv.amap, mBinding, mInfo.FormalMethodInst) with
+                | [[]], retTy when isByrefTy g retTy && mInfo.IsInstance -> true
+                | _ -> false
+            )
+        
+        match getPinnableReferenceMInfo with
+        | Some mInfo ->
+            checkLanguageFeatureError g.langVersion LanguageFeature.ExtendedFixedBindings mBinding
```

Comment:
Shouldn't this check go all the way to the top of the function?

Reply:
You are correct.

Reply:
I think it needs to search for the method either way, so that when the feature is disabled it can still distinguish between these two situations:

```fsharp
[<Struct; IsByRefLike>]
type NonPinnableSpanLike<'T> = { Value: 'T }

let f (x: NonPinnableSpanLike<'T>) =
    // Should always raise FS3207 - "Invalid use of 'fixed'"
    use ptr = fixed x
    NativePtr.get ptr 0
```

```fsharp
open System

// Span has GetPinnableReference()
let f (x: Span<'T>) =
    // When feature enabled, should pass compilation
    // When feature disabled, should raise FS3350 - "Feature is not available in F# X"
    use ptr = fixed x
    NativePtr.get ptr 0
```

Reply:
Done